newIDE/app/src/CompactPropertiesEditor/index.js

```diff
@@ -0,0 +1,791 @@
+      ...ValueFieldCommonProperties,
+    |}
+  | {|
+      valueType: 'color',
+      getValue: Instance => string,
+      setValue: (instance: Instance, newValue: string) => void,
+      ...ValueFieldCommonProperties,
+    |}
+  | {|
+      valueType: 'textarea',
+      getValue: Instance => string,
+      setValue: (instance: Instance, newValue: string) => void,
+      ...ValueFieldCommonProperties,
+    |};
+
+// "Resource" fields are showing a resource selector.
+type ResourceField = {|
+  valueType: 'resource',
+  resourceKind: ResourceKind,
+  fallbackResourceKind?: ResourceKind,
+  getValue: Instance => string,
+  setValue: (instance: Instance, newValue: string) => void,
+  renderLeftIcon?: (className?: string) => React.Node,
+  ...ValueFieldCommonProperties,
+|};
+
+type Title = {|
+  name: string,
+  renderLeftIcon: (className?: string) => React.Node,
+  getValue?: Instance => string,
+  nonFieldType: 'title',
+  defaultValue?: string,
+|};
+
+type SectionTitle = {|
+  name: string,
+  nonFieldType: 'sectionTitle',
+  getValue: typeof undefined,
+|};
+
+type VerticalCenterWithBar = {|
+  name: string,
+  nonFieldType: 'verticalCenterWithBar',
+  child: PrimitiveValueField,
+|};
+
+type ActionButton = {|
+  label: string,
+  disabled: 'onValuesDifferent',
+  getValue: Instance => string,
+  nonFieldType: 'button',
+  getIcon?: ({| fontSize: string |}) => React.Node,
+  onClick: (instance: Instance) => void,
+|};
+
+// A value field is a primitive or a resource.
+export type ValueField = PrimitiveValueField | ResourceField;
+
+// A field can be a primitive, a resource or a list of fields
+export type Field =
+  | PrimitiveValueField
+  | ResourceField
+  | SectionTitle
+  | Title
+  | ActionButton
+  | VerticalCenterWithBar
+  | {|
+      name: string,
+      type: 'row' | 'column',
+      preventWrap?: boolean,
+      title?: ?string,
+      children: Array<Field>,
+    |};
+
+// The schema is the tree of all fields.
+export type Schema = Array<Field>;
+
+type Props = {|
+  onInstancesModified?: Instances => void,
+  instances: Instances,
+  schema: Schema,
+  mode?: 'column' | 'row',
+  preventWrap?: boolean,
+
+  // If set, render the "extra" description content from fields
+  // (see getExtraDescription).
+  renderExtraDescriptionText?: (extraDescription: string) => string,
+  unsavedChanges?: ?UnsavedChanges,
+
+  // Optional context:
+  project?: ?gdProject,
+  resourceManagementProps?: ?ResourceManagementProps,
+|};
+
+const styles = {
+  columnContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  fieldContainer: {
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'baseline',
+  },
+  field: {
+    flex: 1,
+    width: 'auto',
+  },
+  subHeader: {
+    paddingLeft: 0,
+  },
+  container: { flex: 1, minWidth: 0 },
+  separator: {
+    marginRight: -marginsSize,
+    marginLeft: -marginsSize,
+    marginTop: marginsSize,
+    borderTop: '1px solid black',
```

Comment:
just to confirm this color looks ok on all themes? (plus you apply a borderColor from the theme just below)

Reply:
The border color is changed yes, I will add a comment.